lib/msal-node/src/utils/Constants.ts

```diff
@@ -43,6 +43,19 @@ export const ManagedIdentitySourceNames = {
 export type ManagedIdentitySourceNames =
     (typeof ManagedIdentitySourceNames)[keyof typeof ManagedIdentitySourceNames];
 
+/**
+ * Azure Identity SDK defined identifiers for the Managed Identity Sources
+ */
+export const AzureIdentitySdkManagedIdentitySourceNames = {
+    APP_SERVICE: "APP_SERVICE",
+    AZURE_ARC: "ARC",
+    CLOUD_SHELL: "CLOUD_SHELL",
+    IMDS: "DEFAULT_TO_VM",
```

Comment:
The caller would mainly use the const's names rather than their values. Better also name the const same `DEFAULT_TO_VM` here. The meaning we want to convey here is that the new helper `GetManagedIdentitySource()` does not have confidence on whether the underlying environment is a VM, therefore the caller probably want to probe.

Reply:
I added an enum to the list instead as renaming the IMDS enum. Renaming will be a breaking change in case anyone already have a check on this. We do send back the source in Error scenarios. See https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/4761/files/de6bc4bb5cca106e3bc3c9137908e1c3af45f494#diff-13058fddef4867b4d36dca3120b5b961ae64b039b32bcf5ad410a364b36fbfa8R83

Reply:
> Renaming will be a breaking change in case anyone already have a check on this.

Removing existing names would be a breaking change. But you may always add a new alias for existing name, and then the old name can be considered deprecated.

Reply:
In regards to the original comment, this string enum is not public api. Azure Identity SDK doesn't see the keys, only the values, which are their requested strings. I think the name of the enum is clear enough, for the developers sake.

Reply:
Implemented Ray's feedback, per offline discussion.

Reply:
@neha-bhargava, I didn't rename any existing constants. I made a new string enum just for the api that Azure Identity SDK consumes.